src/redux/slices/user.ts

```diff
@@ -21,165 +33,177 @@ import {
+    return formatAssets(assets, icpPrice);
+  } catch (e: any) {
+    console.log('getBalance error', e);
+    return rejectWithValue(e.message);
   }
-);
+});
 
-export const getTransactions = createAsyncThunk(
-  'user/getTransactions',
-  async (params, { rejectWithValue }) => {
+export const getNFTs = createAsyncThunk<
+  Collection[],
+  { refresh: boolean } | undefined,
+  { rejectValue: string }
+>('user/getNFTs', async (params, { rejectWithValue }) => {
+  if (ENABLE_NFTS) {
     try {
-      const { icpPrice } = params;
       const instance = KeyRing.getInstance();
-      const currentWalletId = instance?.currentWalletId;
-      const currentWallet = instance?.state?.wallets[currentWalletId];
-      const response = await instance?.getTransactions();
-      let parsedTrx =
-        response?.transactions?.map(
-          formatTransaction(icpPrice, currentWallet)
-        ) || [];
-
-      if (!ENABLE_NFTS) {
-        parsedTrx = parsedTrx.filter(
-          item =>
-            !(
-              item?.symbol === 'NFT' ||
-              item?.details.canisterId === JELLY_CANISTER_ID
-            )
-        );
-      }
-
-      return parsedTrx;
-    } catch (e) {
-      return rejectWithValue({
-        error: e.message,
+      const collections = await instance.getNFTs({
+        refresh: params?.refresh,
       });
+      return (collections || []).map(collection =>
+        recursiveParseBigint(collection)
+      );
+    } catch (e: any) {
+      return rejectWithValue(e.message);
     }
   }
-);
+  return [];
+});
+
+export const getTransactions = createAsyncThunk<
+  Transaction[],
+  { icpPrice: number },
+  { rejectValue: string }
+>('user/getTransactions', async (params, { rejectWithValue }) => {
+  try {
+    const { icpPrice } = params;
+    const instance = KeyRing.getInstance();
+    const currentWalletId = instance?.currentWalletId;
+    const state = await instance?.getState();
+    const currentWallet = state.wallets[currentWalletId];
+    const response = await instance?.getTransactions({});
+    let parsedTrx =
+      response?.transactions?.map(formatTransaction(icpPrice, currentWallet)) ||
+      [];
+
+    if (!ENABLE_NFTS) {
+      parsedTrx = parsedTrx.filter(
+        item =>
+          !(
+            item?.symbol === 'NFT' ||
+            item?.details.canisterId === JELLY_CANISTER_ID
+          )
+      );
+    }
+
+    return parsedTrx;
+  } catch (e: any) {
+    return rejectWithValue(e.message);
+  }
+});
 
 export const transferNFT = createAsyncThunk(
   'user/transferNFT',
-  async (params, { rejectWithValue, dispatch }) => {
+  async (
+    params: {
+      to: string;
+      nft: CollectionToken;
+      icpPrice: number;
+      onEnd?: () => void;
+    },
+    { rejectWithValue, dispatch }
+  ) => {
     try {
       const { to, nft, icpPrice, onEnd } = params;
       const instance = KeyRing.getInstance();
       const response = await instance?.transferNFT({
         to,
+        // @ts-ignore
```

Comment:
is this ok?

Reply:
this is the problem, i dont know what can i do 😆 
<img width="677" alt="Screen Shot 2022-10-13 at 16 07 24" src="https://user-images.githubusercontent.com/25931366/195686972-3a79d3cc-063e-46e0-b031-6548905e8b55.png">